lib/msf/core/modules/metadata/search.rb

```diff
@@ -192,19 +211,30 @@ def is_match(params, module_metadata)
           break if match
         end
         # Filter this module if no matches for a given keyword type
-        if mode == 0 and not match
+        if mode == SearchMode::INCLUDE and not match
           return false
         end
       end
       # Filter this module if we matched an exclusion keyword (-value)
-      if mode == 1 and match
+      if mode == SearchMode::EXCLUDE and match
         return false
       end
     end
 
     true
   end
 
+  def as_regex(search_term)
+    # Convert into a case-insensitive regex
+    utf8_buf = search_term.dup.force_encoding('UTF-8')
+    if utf8_buf.valid_encoding?
+       Regexp.new(Regexp.escape(utf8_buf), Regexp::IGNORECASE)
+    else
+      # If the encoding is invalid, default to a regex that matches anything
+      //
```

Comment:
Should this be a `Regex` object aswell like we have above?

Reply:
@dwelch-r7 Regex is a first class citizen in Ruby's syntax, so `/.../` is actually just syntactic sugar for Regex.new.
However you can use `Regexp.new` if you want to convert a string to a regex.

So in this case, an empty regex of `//` is perfectly cromulent

Reply:
ah I didn't know that, neat! I'm for sure glad the comment is there though, if nothing else but to remind me that # is a comment in ruby not // 😄